gpflow/base.py

```diff
@@ -14,8 +14,8 @@
 TensorLike = (
     object  # Union[tf.Tensor, tf.Variable, np.ndarray], but doesn't work with multipledispatch
 )
-Transform = tfp.bijectors.Bijector
-Prior = tfp.distributions.Distribution
+Transform = Type[tfp.bijectors.Bijector]
+Prior = Type[tfp.distributions.Distribution]
```

Comment:
are you sure `Type` is right here? I believe `Type[<class>]` is just `type`, the metaclass as that's the type of a class (now say that 10 times fast). You mentioned `Union` before

Reply:
nope, I got that wrong. gimme a minute

Reply:
given `o: Type[tfp.bijectors.Bijector]`, `o` has to be either the class `tfp.bijectors.Bijector` or a subclass of it, but _not an instance_ of type `tfp.bijectors.Bijector`

Reply:
My understanding of https://docs.python.org/3/library/typing.html#typing.Type is that `Type[<class>]` will match `<class>` and subclasses of `class`, which seems to be the desired behaviour here. I can change it to `Union` if you prefer?

Reply:
see line 106 ... `inverse_log_det_jacobian` is an instance method not a class method so `self.transform` must be an instance of a `tfp.bijectors.Bijector`. It's also an instance of a `Transform`, so that means `Transform ~ tfp.bijectors.Bijector` not `~ Type[tfp.bijectors.Bijector]`?

Reply:
(this is why we need mypy runs on gpflow)

Reply:
I'm pretty sure `Type` is one level higher up the is-a chain than we need. I'm reasonably confident `Union` would work (as odd as it looks). If you're OK to do that, yes please

Reply:
Yep, no problem.